private/my-local-model/src/endpoint/EndpointParameters.ts

```diff
@@ -5,25 +5,53 @@ import type { Endpoint, EndpointParameters as __EndpointParameters, EndpointV2,
  * @public
  */
 export interface ClientInputEndpointParameters {
+  clientContextParams?: {
+    apiKey?: string | undefined | Provider<string | undefined>;
+    customParam?: string | undefined | Provider<string | undefined>;
+  };
   endpoint?: string | Provider<string> | Endpoint | Provider<Endpoint> | EndpointV2 | Provider<EndpointV2>;
+  apiKey?: string | undefined | Provider<string | undefined>;
+  customParam?: string | undefined | Provider<string | undefined>;
 }
 
-export type ClientResolvedEndpointParameters = Omit<ClientInputEndpointParameters, "endpoint"> & {
+export type ClientResolvedEndpointParameters = Omit<
+  ClientInputEndpointParameters,
+  "endpoint" | "clientContextParams"
+> & {
   defaultSigningName: string;
+  clientContextParams: {
+    apiKey?: string | undefined | Provider<string | undefined>;
+    customParam?: string | undefined | Provider<string | undefined>;
+  };
 };
 
+/**
+ * @internal
+ */
+const clientContextParamDefaults = {
+  apiKey: "default-api-key",
+  customParam: "default-custom-value",
+} as const;
+
 export const resolveClientEndpointParameters = <T>(
   options: T & ClientInputEndpointParameters
 ): T & ClientResolvedEndpointParameters => {
   return Object.assign(options, {
+    apiKey: options.apiKey ?? "default-api-key",
```

Comment:
This is the main point of the fix, but this shouldn't appear here, since it will conflict with the same key potentially be added by the api key auth scheme.